shared/src/commonMain/kotlin/db/repository/reservation/RemoteReservationRepository.kt

```diff
@@ -1,44 +1,89 @@
 package db.repository.reservation
 
-import db.chargehub.Reservation
 import db.database.reservation.ReservationDatabase
 import db.database.reservation.ReservationDatabaseWrapper
 import db.networking.request.CreateReservationRequest
+import db.networking.request.GetReservationResponse
 import db.repository.GenericRepository
+import dev.gitlive.firebase.Firebase
+import dev.gitlive.firebase.firestore.DocumentSnapshot
+import dev.gitlive.firebase.firestore.firestore
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.flowOf
+import kotlinx.datetime.LocalDateTime
 import org.koin.core.component.inject
 
 class RemoteReservationRepository :
-    GenericRepository<CreateReservationRequest, Reservation, ReservationDatabase> {
+    GenericRepository<CreateReservationRequest, GetReservationResponse, ReservationDatabase> {
+
+    companion object {
+        const val COLLECTION_NAME = "RESERVATIONS"
+    }
+
+    private val firestore = Firebase.firestore
+
     override val database: ReservationDatabase
         get() = inject<ReservationDatabaseWrapper>().value.database
 
-    override suspend fun fetchAll(): List<Reservation> {
-        return database.getAll()
+    override suspend fun fetchAll(): List<GetReservationResponse> {
+        return try {
+            val data = firestore.collection(COLLECTION_NAME)
+                .get().documents.map<DocumentSnapshot, GetReservationResponse> { it.data() }
+            data.forEach {
+                database.update(
+                    it.id, CreateReservationRequest(
+                        userId = it.userId,
+                        chargeHubId = it.chargeHubId,
+                        carId = it.carId,
+                        startTime = LocalDateTime.parse(it.startTime),
+                        endTime = LocalDateTime.parse(it.endTime)
+                    )
+                )
+            }
+            data
+        } catch (e: Exception) {
+            e.printStackTrace()
+            listOf()
```

Comment:
So now when a call fails the user is not notified and he gets an empty list if i'm not mistaken? Not a priority for now but it would be cool to implement error handling by using a snackbar in android and a dialog in iOS. Add it as a ticket to the backlog

Reply:
The user isn't notified, it will just return an empty list of reservations, and print the stack trace. I used it for debugging on why it failed to retrieve reservations out of the database, and figured it'd be useful to developers to see an error in the console. (this was due to something wrong in our firebase). If you prefer I can remove it or I can make it so it doesn't catch the error and simply crash the app.

Reply:
You may keep it as is for now. It's a possibility to create a wrapping data class like FirebaseRepsonse<T> that can have a error message that you can show in a snackbar in the UI, just so u know what options there are to handle this